node_modules/col-activities/tests/test-daily.js

```diff
@@ -0,0 +1,788 @@
+      assert.ok(!err);
+      assert.ok(course);
+      return callback(course);
+    });
+  };
+
+  /**
+   * Given a set of collected emails, get the one matching a specific email address
+   *
+   * @param  {Object[]}   emails    Collected emails (as per ActivitiesTestUtil.assertDailyNotifications)
+   * @param  {String}     to        The email address to look for
+   * @return {Object}               The matching email or `undefined` if it could not be found
+   * @api private
+   */
+  var getEmail = function(emails, to) {
+    return _.find(emails, function(email) {
+      return (email.email.to === to);
+    });
+  };
+
+  /**
+   * Assert that an email contains references to (and only to) a set of assets
+   *
+   * @param  {Object[]}   email     Collected email (as per ActivitiesTestUtil.assertDailyNotifications)
+   * @param  {Asset[]}    assets    The set of assets that should be in the email
+   * @api private
+   */
+  var assertEmailContainsAssets = function(email, assets) {
+    // Get all the asset ids in the email
+    var re = new RegExp(getAssetLibraryBaseUrl(email.course) + '\\?col_asset=(\d+)');
+    var assetIds = getIdsFromURLs(email, re);
+
+    // Ensure each asset in the email is supposed to be referenced
+    _.each(assetIds, function(id) {
+      assert.ok(_.find(assets, {'id': id}));
+    });
+
+    // Ensure all assets are present both by title and URL
+    _.each(assets, function(asset) {
+      // Assert the title of the asset is present in the email
+      assert.ok(email.email.html.indexOf(asset.title) > -1);
+
+      // Assert a full link to the asset is present
+      var expectedLink = getAssetLibraryBaseUrl(email.course) + '?col_asset=' + asset.id;
+      assert.ok(email.email.html.indexOf(expectedLink) > -1);
+    });
+  };
+
+  /**
+   * Assert that an email contains references to (and only to) a set of users
+   *
+   * @param  {Object[]}   email     Collected email (as per ActivitiesTestUtil.assertDailyNotifications)
+   * @param  {User[]}     users    The set of users that should be in the email
+   * @api private
+   */
+  var assertEmailContainsUsers = function(email, users) {
+    // Get all the user ids in the email
+    var re = new RegExp(getAssetLibraryBaseUrl(email.course) + '\\?col_user=(\d+)');
+    var userIds = getIdsFromURLs(email, re);
+
+    // Ensure each user in the email is supposed to be referenced
+    _.each(userIds, function(id) {
+      assert.ok(_.find(users, {'id': id}));
+    });
+
+    // Ensure all users are present both by full name and URL
+    _.each(users, function(user) {
+      // Assert the name of the user is present in the email
+      assert.ok(email.email.html.indexOf(user.canvas_full_name) > -1);
+
+      // Assert a full link to the user's asset library is present
+      var expectedLink = getAssetLibraryBaseUrl(email.course) + '?col_user=' + user.id;
+      assert.ok(email.email.html.indexOf(expectedLink) > -1);
```

Comment:
It might be worth creating a `ensurePresent(string, fragment)` function to avoid doing the `.indexOf() > -1` everywhere